src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props

```
@@ -351,18 +351,31 @@
 	</CreateProperty>
 
   </Target>
-
+  
   <Target Name="_InjectAdditionalFiles" BeforeTargets="GenerateMSBuildEditorConfigFileShouldRun;GenerateMSBuildEditorConfigFileCore" DependsOnTargets="_FillSourceGeneratorItemGroups">
-	<ItemGroup>
-	  <AdditionalFiles Include="@(Page)" SourceItemGroup="Page" />
-	  <AdditionalFiles Include="@(ApplicationDefinition)" SourceItemGroup="ApplicationDefinition" />
-	  <AdditionalFiles Include="@(PRIResource)" SourceItemGroup="PRIResource" />
-	  <AdditionalFiles Include="@(TSBindingAssemblySource)" SourceItemGroup="TSBindingAssemblySource" />
-
-	  <_AdditionalFilesCleanup Include="@(AdditionalFiles)" />
-	  <AdditionalFiles Remove="@(_AdditionalFilesCleanup)" />
-	  <AdditionalFiles Condition="'@(_AdditionalFilesCleanup)'!=''" Include="@(_AdditionalFilesCleanup->'$([System.IO.Path]::GetFullPath('%(_AdditionalFilesCleanup.Identity)'))')" />
-	</ItemGroup>
+
+    <PropertyGroup>
+      <_Pages>@(Page)</_Pages>
+      <_ApplicationDefinitions>@(ApplicationDefinition)</_ApplicationDefinitions>
+      <_PRIResources>@(PRIResource)</_PRIResources>
+      <_TSBindingAssemblySources>@(TSBindingAssemblySource)</_TSBindingAssemblySources>
+    </PropertyGroup>
+    
+    <ItemGroup Label="Add SourceItemGroup metadata to existing additional files">
+      <AdditionalFiles SourceItemGroup="Page" Condition="$([System.String]::new($(_Pages)).Contains(%(AdditionalFiles.FullPath)))" />
```

I wonder what's the performance impact of this, since it'll recreate a string for all the pages of the app. Can't it be 
```suggestion
      <AdditionalFiles SourceItemGroup="Page" Condition="$(_Pages.Contains(%(AdditionalFiles.FullPath)))" />
```



That would avoid msbuild to re-create the full string for each AdditionalFile already present?